### ISSUE TITLE
This is a "no need to verify" hacking tool

### ISSUE THREAD

**Issue by DarkmoonRabbit** [azerothcore/azerothcore-wotlk#465]
**Description**:
Known GM account, do not need to use the correct password to log into the game.
Perhaps the login validation has occurred bug!

**Tool source code**: https://github.com/namreeb

**Solution?**:

```
src/server/authserver/Server/AuthSocket.cpp | 5 ++++-
1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/server/authserver/Server/AuthSocket.cpp b/src/server/authserver/Server/AuthSocket.cpp
index 8ad12f7..181e8ea 100644
--- a/src/server/authserver/Server/AuthSocket.cpp
+++ b/src/server/authserver/Server/AuthSocket.cpp
@@ -1,4 +1,4 @@
-/*
+﻿/*
  * Copyright (C)
  * Copyright (C)
  *
@@ -573,6 +573,9 @@ bool AuthSocket::_HandleLogonProof()
         return true;
     }

+        if ((A % N).isZero())
+                return false;
+
     SHA1Hash sha;
     sha.UpdateBigNumbers(&A, &B, NULL);
     sha.Finalize();
-- 
2.10.1.windows.1
```

**Comment by Yehonal** [comment]
Hi! could you open a PR? Does this bug/security issue exists on TC too? if yes could you link us the issue?

Thank you

**Comment by starrheld** [comment]
What's missing; hence your "possible fix"

https://github.com/starrheld/azerothcore-wotlk/blob/8a5310788cc2353ed2104b638483acb5c753f839/src/authserver/Server/AuthSocket.cpp#L560

**Comment by DarkmoonRabbit** [comment]
@starrheld I have not tested it, it's just an idea.
@Yehonal Sorry, I will not use github, I only know how to clone. all of the 335 versions have this bug.

**Comment by Yehonal** [comment]
We already have the fix for it. It's an old bug fixed various months ago on
both AC and TC. Have you tried with latest AC 1.x?

Thanks anyway

Il 19/Apr/2017 16:26, "DarkmoonRabbit" <notifications@github.com> ha
scritto:

**Comment by DarkmoonRabbit** [comment]
@Yehonal My AC version is up to date! And the hacker tool author also upgraded the version. I tested, AC died... https://github.com/namreeb

**Comment by sucofog** [comment]
I don't think so.

**Comment by Yehonal** [comment]
@DarkmoonRabbit so sorry but if the vulnerability is there, then your fix is not the answere since, as starrheld said, we already have that check: 

https://github.com/azerothcore/azerothcore-wotlk/blob/a67ceca682a813303de21d8f5a362134004315b6/src/authserver/Server/AuthSocket.cpp#L560

we imported both fixes:

https://github.com/cmangos/mangos-classic/commit/74d51cf70d67f6d4a47321a4226e7473cb8e2601
https://github.com/cmangos/mangos-classic/commit/0d2b7e38c886ddd6828cfa75e2daba5121467383

but maybe @namreeb can say us something more

**Comment by namreeb** [comment]
Your code appears to be safe from both exploits.

**Comment by Yehonal** [comment]
@namreeb i've tried on a test server and your exploit seems to work unfortunately.

**Comment by sucofog** [comment]
It does not works for latest AC revision for me, tested on my own server:

![image](https://cloud.githubusercontent.com/assets/734950/25193350/195919e8-2537-11e7-825a-c3cb282f78c1.png)

![image](https://cloud.githubusercontent.com/assets/734950/25193355/204ebdde-2537-11e7-90b6-b2266a380e3d.png)

![image](https://cloud.githubusercontent.com/assets/734950/25193363/2e38e94c-2537-11e7-9658-9ff91bbcfca2.png)


Tested with both methods

**Comment by namreeb** [comment]
@Yehonal I don't know what to say.  Maybe your binary is out of date?  The code on github (here: https://github.com/azerothcore/azerothcore-wotlk/blob/a67ceca682a813303de21d8f5a362134004315b6/src/authserver/Server/AuthSocket.cpp) sure appears safe to me.

**Comment by Rochet2** [comment]
> I tested, AC died...

@DarkmoonRabbit What do you mean AC died? Crashed?

**Comment by DarkmoonRabbit** [comment]
@Rochet2 AC died == 'hack tool Effective'

**Comment by Mycadi** [comment]
@Yehonal  AC 1.x   and  AC master   Not the same?

**Comment by Yehonal** [comment]
no, infact @starrheld we've to import your commit https://github.com/azerothcore/azerothcore-wotlk/commit/a67ceca682a813303de21d8f5a362134004315b6 
on 1.x branch since it's a security fix https://github.com/azerothcore/azerothcore-wotlk/wiki/Project-Versioning

**Comment by Yehonal** [comment]
I've cherry-picked the fix on branch 1.x , if you're using that please update the revision and tell us if it has been fixed.

Thank you

**Comment by Yehonal** [comment]
however @DarkmoonRabbit , please next time just follow issue guidelines where you need to specify branch used

**Comment by DarkmoonRabbit** [comment]
@Yehonal ok,thank you! Until today, I only know that the original Github has a lot of "branches", I would like to ask 1.x and master who is best? my version is master.

**Comment by Sworken** [comment]
Confirmed! AC has vulnerability, on my server are using this :(  
what is the solution?

I'll try this for now: https://github.com/azerothcore/azerothcore-wotlk/commit/a67ceca682a813303de21d8f5a362134004315b6

**Comment by Franklampardst** [comment]
Guys i tested after https://github.com/starrheld/azerothcore-wotlk/blob/8a5310788cc2353ed2104b638483acb5c753f839/src/authserver/Server/AuthSocket.cpp#L560 not Working that HACK!!!

**Comment by Krilliac** [comment]
If this is still an issue (For what ever reason), this hack is also detectable by Warden.

INSERT INTO `warden_checks` VALUES
(787,243,'','',4618640,6,'558BEC83EC50','Auth Bypass'),
(788,243,'','',11685160,3,'80C98C','Auth Bypass'),
(789,243,'','',9226342,2,'7548','Auth Bypass'),
(790,243,'','',7016192,6,'8B0DEC9CC700','Auth Bypass'),
(791,243,'','',10295321,5,'A3284DB200','Auth Bypass'),
(792,243,'','',10126304,6,'558BEC81EC5C','Auth Bypass');

**Comment by Nefertumm** [comment]
This should be closed, the problem was fixed on a67ceca682a813303de21d8f5a362134004315b6. :)
